### ISSUE TITLE
[java] [False positive] PMD fails to recognize method call when type passed in is var.

### ISSUE THREAD

**Issue by Fernal73** [pmd/pmd#2143]
**Affects PMD Version:** 
Running against 6.19.0.

**Rule:**
UnusedPrivateMethod.

**Description:**
https://stackoverflow.com/q/58529020/3924108
This post on StackOverflow (over a month ago) describes a bug that has not yet been fixed. Why is there no process to move issues reported there? The least that could be done is request the OP to do so.

Note: I am in no way endorsing moving PMD support to StackOverflow.

https://meta.stackexchange.com/q/3966/509478

**Code Sample demonstrating the issue:**

```java
package pmdtests;

import java.util.List;

public final class TestVar {

private TestVar() {
  throw new IllegalStateException("Private constructor.");
}

private static int addProducts(List<String> prods) {
    var rowAdder = new RowAdder();
    rowAdder.add(prods);
    processDocument(
                    rowAdder);
    return rowAdder.intValue();
}

private void processDocument(RowAdder rowAdder) {
  // do nothing
  //
System.out.println("Processing " + rowAdder);
}

private static class RowAdder {
public void addProducts(List<String> prods) {
  System.out.println("Adding " + prods);
}

public int intValue() {
return 5;
} 
}
}
```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

**Comment by Fernal73** [comment]
It appears that I was looking at the top class method, not the inner static class. Closing this issue.
